rust/client/src/ws/mod.rs

```diff
@@ -8,43 +9,46 @@ use tokio::sync::mpsc::Sender;
 use tokio_tungstenite::tungstenite::protocol::Message;
 use tokio_tungstenite::{connect_async, tungstenite::Utf8Bytes};
 
-use crate::{BACKPACK_WS_URL, BpxClient, DEFAULT_WINDOW, now_millis};
+use crate::{BpxClient, DEFAULT_WINDOW, Error, now_millis};
 
 impl BpxClient {
     /// Subscribes to a private WebSocket stream and sends messages of type `T` through a transmitter channel.
-    pub async fn subscribe<T>(&self, stream: &str, tx: Sender<T>)
+    pub async fn subscribe<T>(&self, stream: &str, tx: Sender<T>) -> Result<()>
     where
         T: DeserializeOwned + Send + 'static,
     {
         self.internal_subscribe(&[stream], tx).await
     }
 
     /// Subscribes to multiple private WebSocket streams and sends messages of type `T` through a transmitter channel.
-    pub async fn subscribe_multiple<T>(&self, stream: &[&str], tx: Sender<T>)
+    pub async fn subscribe_multiple<T>(&self, stream: &[&str], tx: Sender<T>) -> Result<()>
     where
         T: DeserializeOwned + Send + 'static,
     {
         self.internal_subscribe(stream, tx).await
     }
 
-    async fn internal_subscribe<T>(&self, stream: &[&str], tx: Sender<T>)
+    async fn internal_subscribe<T>(&self, stream: &[&str], tx: Sender<T>) -> Result<()>
     where
         T: DeserializeOwned + Send + 'static,
     {
         let timestamp = now_millis();
         let window = DEFAULT_WINDOW;
         let message = format!("instruction=subscribe&timestamp={timestamp}&window={window}");
+        let Some(signing_key) = &self.signing_key else {
```

Comment:
Not all WS streams require auth, only those that are prefixed with `"account."`

I covered this in https://github.com/backpack-exchange/bpx-api-client/pull/62, could copy from there